### PR TITLE
Refactor storage of supplemental attributes and time series

### PR DESCRIPTION
**Pull request by daniel-thom**
Remove supplemental_attributes_container and time_series_container from component structs and make them implement supports_supplemental_attributes and supports_time_series instead.

### CODE REVIEW

File: src/PowerSystems.jl [hunk 1]
```diff
@@ -319,11 +319,14 @@ export remove_supplemental_attribute!
 export remove_supplemental_attributes!
 export get_supplemental_attribute
 export get_supplemental_attributes
+export list_supplemental_attributes
```
Comment (jd-lara) [inline src/PowerSystems.jl:322]:
```suggestion
export get_supplemental_attributes
```

File: test/test_outages.jl [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
     for gen in (gen1, gen2)
         for type in (GeometricDistributionForcedOutage, PlannedOutage, GeographicInfo)
-            attributes = collect(get_supplemental_attributes(type, gen))
+            attributes = list_supplemental_attributes(type, gen)
```
Comment (jd-lara) [inline test/test_outages.jl:24]:
```suggestion
            attributes = get_supplemental_attributes(type, gen)
```

File: test/test_outages.jl [hunk 3]
```diff
@@ -39,14 +39,14 @@
         ),
     ) == 1
     @test length(
-        get_supplemental_attributes(
+        list_supplemental_attributes(
```
Comment (jd-lara) [inline test/test_outages.jl:42]:
```suggestion
        get_supplemental_attributes(
```

Comment (daniel-thom) [reply test/test_outages.jl:42]: Yeah, I changed all of these this morning and have been waiting for the CostCurve fix.

**Pull request merged**